src/widgets/PieChart.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2018
+ */
+
+import React, { useState, useEffect } from 'react';
+import { View, StyleSheet } from 'react-native';
+import PropTypes from 'prop-types';
+import { VictoryLabel, VictoryPie } from 'victory-native';
+import { APP_FONT_FAMILY, GREY } from '../globalStyles';
+
+/**
+ * A charting widget for displaying Pie chart report.
+ *
+ * @prop  {string}       id      The report ID.
+ * @prop  {string}       title   The title of the report.
+ * @prop  {string}       type    The type of chart to use to display the report,
+ *                               options are BarChart, LineChart and PieChart.
+ * @prop  {data}         array   An array of {x, y} datapoints to plot.
+ * @prop  {width}        number  The width of the parent container.
+ * @prop  {height}       number  The height of the parent continer.
+ */
+
+export const ReportChart = ({ report: initialReport }, props) => {
+  const [report, setReport] = useState(initialReport);
+  const [width, setWidth] = useState(null);
+  const [height, setHeight] = useState(null);
+
+  // Same affect as componentDidUpdate
+  useEffect(() => {
+    if (report.id !== props.report.id) setReport(props.report);
+  });
```

Comment:
The previous component was trying to manage being passed different reports and re-rendering. I.e. when you click on the side bar it would render a new chart.

In this case, your `PieChart` is a single component that won't really change, so I don't think this is neccessary at all.

Reply:
Thanks, I took it out!